spec/controllers/affiliate_controller_spec.rb

```diff
@@ -2,4 +2,140 @@
 
 RSpec.describe AffiliateController, type: :controller do
 
+
+  # Return the minimal set of attributes required to create a valid Affiliate.
+  # As you add validations to Affiliate, be sure to adjust the attributes here as well.
+  let(:valid_attributes) {
+    skip("Add a hash of attributes valid for your model")
+  }
+
+  let(:invalid_attributes) {
+    skip("Add a hash of attributes invalid for your model")
+  }
+
+  # Return the minimal set of values that should be in the session
+  # in order to pass any filters (e.g. authentication) defined in
+  # AffiliatesController.
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "assigns all affiliates as @affiliates" do
+      affiliate = Affiliate.create! valid_attributes
+      get :index, params: {}, session: valid_session
+      expect(assigns(:affiliates)).to eq([affiliate])
+    end
```

Comment:
This test, and all of the other ones that use `affiliate = Affiliate.create! valid_attributes`, is/are broken. If you comment out line 24 you'll see that this still passes, which is bad! Instead, create a new affiliate factory at `spec/factories/affiliate.rb` (base it off of `spec/factories/student.rb`). Then model your specs off of the student controller spec, like this:

```
      describe 'GET #index' do
        it 'populates an array of all students' do
          students = [create(:student)]
          get :index
          expect(assigns(:students)).to eq(students)
        end

        it 'renders the :index view' do
          get :index

          expect(response).to render_template :index
        end
      end
```

Also, you can use a `before` block to use a valid session for every request, also in the student controller spec. We can go over this on Sunday if you like!